corgi/tasks/rhel_compose.py

```diff
@@ -38,36 +38,16 @@ def save_compose(stream_name) -> None:
     logger.info("Created %s new relations for stream %s", no_of_relations, stream_name)
 
 
-def get_builds_by_compose(compose_names):
-    relations_query = (
-        ProductComponentRelation.objects.filter(
-            external_system_id__in=compose_names,
-            type=ProductComponentRelation.Type.COMPOSE,
-        )
-        .values_list("build_id", flat=True)
-        .distinct()
-    )
-    fetch_modular_builds(relations_query)
-
-
-def get_builds_by_stream(stream_name):
-    relations_query = (
-        ProductComponentRelation.objects.filter(
-            product_ref=stream_name,
-            type=ProductComponentRelation.Type.COMPOSE,
-        )
-        .values_list("build_id", flat=True)
-        .distinct()
+@app.task(base=Singleton, autoretry_for=RETRYABLE_ERRORS, retry_kwargs=RETRY_KWARGS)
+def get_builds(compose_names: list[str] = None, stream_name: str = None) -> None:
+    """Get compose build IDs, optionally for only a particular stream or set of composes"""
+    relations_query = ProductComponentRelation.objects.filter(
+        type=ProductComponentRelation.Type.COMPOSE
     )
-    fetch_modular_builds(relations_query)
-
+    if compose_names:
+        relations_query = relations_query.filter(external_system_id__in=compose_names)
+    elif stream_name:
+        relations_query = relations_query.filter(product_ref=stream_name)
 
-def get_all_builds():
-    relations_query = (
-        ProductComponentRelation.objects.filter(
-            type=ProductComponentRelation.Type.COMPOSE,
-        )
-        .values_list("build_id", flat=True)
-        .distinct()
-    )
-    fetch_modular_builds(relations_query)
+    relations_build_ids = relations_query.values_list("build_id", flat=True).distinct()
```

Comment:
Everywhere I see the use of distinct I think we should add an order_by call as well. I've found it's necessary for distinct to work correctly with Django and Postgres.

Perhaps we should add an updated filter here as well and only update composes build which might have changed since the last run? See for example the slow_fetch_unprocess_cdn_relatiions task.

```
        created_during = timezone.now() - timedelta(days=8)
        relations_query = relations_query.filter(created_at__gte=created_during)
```

Reply:
Hmm, that's actually a really big gotcha I wasn't aware of. I've tested e.g. ```.values_list("some_field", flat=True).distinct()``` and it seemed to work OK. But that was for very little data, so maybe a non-trivial number of records in the DB runs into that ordering bug. I'll file a follow-up bug - can you send me a link to any docs you know of describing this behavior? Here or on Gchat or in the ticket is fine - or if you haven't seen mentioned in docs that's fine too.